---
id: no-duplicate-dt
description: No duplicate names in `<dl>`
---

# `no-duplicate-dt`

No duplicate names in `<dl>`.

❌ Examples of **incorrect** code for this rule

```html
<todo>Write incorrect codes</todo>
```

✅ Examples of **correct** code for this rule

```html
<todo>Write correct codes</todo>
```
